subt_ign/src/GameLogicPlugin.cc

```diff
@@ -1092,6 +1150,77 @@ GameLogicPluginPrivate::UpdateScoreFiles() const
   score << totalScore << std::endl;
   score.flush();
 
+  // output full log to a yml file
+  // 1. Number of artifacts found.
+  // 2. Robot count
+  // 3. Unique robot count (for example, a team of two X1_SENSOR_CONFIG_1
+  // robots would have a unique count of 1).
+  // 4. Total simulation time.
+  // 5. Total real time.
+  // 6. Number of artifact report attempts
+  // 7. Number of duplicate artifact reports.
+  // 8. Closest artifact report:
+  //      a. Distance in meters.
+  //      b. True position of the artifact.
+  //      c. Reported position.
+  //      d. Artifact type and name.
+  // 9. First artifact report time.
+  // 10. Last artifact report time.
+  // 11. Mean time between success reports
+  YAML::Emitter out;
+  out << YAML::BeginMap;
+  out << YAML::Key << "artifacts_found";
+  out << YAML::Value << this->foundArtifacts.size();
+  out << YAML::Key << "robot_count";
+  out << YAML::Value << this->robotNames.size();
+  out << YAML::Key << "unique_robot_count";
+  out << YAML::Value << this->robotSourceFilePaths.size();
+  out << YAML::Key << "sim_time";
+  out << YAML::Value << simElapsed;
+  out << YAML::Key << "real_time";
+  out << YAML::Value << realElapsed;
+  out << YAML::Key << "artifact_report_count";
+  out << YAML::Value << this->reportCount;
+  out << YAML::Key << "duplicate_report_count";
+  out << YAML::Value << this->duplicateReportCount;
+
+  std::stringstream artifactPos;
+  artifactPos << std::get<2>(this->closestReport);
+  std::stringstream reportedPos;
+  reportedPos << std::get<3>(this->closestReport);
+
+  out << YAML::Key << "closest_artifact_report";
+  out << YAML::Value << YAML::BeginMap
+      << YAML::Key << "name"
+      << YAML::Value << std::get<0>(this->closestReport)
+      << YAML::Key << "type"
+      << YAML::Value << std::get<1>(this->closestReport)
+      << YAML::Key << "true_pos"
+      << YAML::Value << artifactPos.str()
+      << YAML::Key << "reported_pos"
+      << YAML::Value << reportedPos.str()
+      << YAML::Key << "distance"
+      << YAML::Value << std::get<4>(this->closestReport)
```

Comment:
Output a value of `-1` instead of `inf` when no artifacts have been reported.

Reply:
done. c65170d